### PR TITLE
Add debounce on every resize listener

### PR DESCRIPTION
**Pull request by louis-bompart**
Resize events are sometimes not triggered in SFINT, because the reflow of the document can take too long, making the outer window well resized but not the inner, nor the div in the document.
That can cause the responsive components breakpoints to not be triggered.

To fix this issue, I propose the following PR, based on the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/Events/resize).

In a nutshell, I'm waiting for the resize event to stop being triggered for at least 250ms before actually calling the listener.

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: unitTests/ui/ResponsiveComponents/ResponsiveComponentsManagerTest.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export function ResponsiveComponentsManagerTest() {
       setTimeout(() => {
```
Comment (samisayegh) [inline unitTests/ui/ResponsiveComponents/ResponsiveComponentsManagerTest.ts]: It would be good to convert this test to use jasmine.clock rather than a real setTimeout to avoid slowing down the tests. There is an example in `FacetSettingsTest.ts`.
